jaxrs-spec/src/main/asciidoc/chapters/resources/_declaring_method_capabilities.adoc

```diff
@@ -126,3 +129,62 @@ the media type `application/xml` given its higher qs-value. Note that
 qs-values, just like q-values, are relative and as such are only
 comparable to other qs-values within the same `@Produces` annotation
 instance. For more information see <<determine_response_type>>.
+
+[[consuming_multipart_formdata]]
+==== Special handling for consuming multipart/form-data media types
+
+The `multipart/form-data` media type enables requests to send multiple
+entities (parts) as a single entity. Each part contains its own set of
+headers, media type and content. There are a few different ways to
+consume these requests in a resource method. First, a resource method
+may contain an entity parameter of type, `List<Part>`. For example:
+
+[source,java]
+----
+@Path("widgets3")
+public class WidgetsResource3 {
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response postWidget(List<Part> parts) {
+        for (Part part : parts) {
+            String name = part.getName();
+            Optional<String> fileName = part.getFileName();
+            InputStream is = part.getContent();
+            MultivaluedMap<String, String> partHeaders = part.getHeaders();
+            MediaType mediaType = part.getMediaType();
+            doSomethingWithPart(name, fileName, is, partHeaders, mediaType);
+        }
+        return Response.ok().build();
+    }
+
+}
+----
+
+Another approach is to use `@FormParam` parameters where the value in the
+annotation corresponds to the name of the part. The parameter type may be
+a `jakarta.ws.rs.ext.Part`, a `java.io.InputStream` or a `String`. Here is
```

Comment:
I wonder if we should really restrict the type to `Part`, `InputStream` and `String` as the API now provides `Part.getContent(Class<T>)` which effectively allows using MBRs to get other types. Why not support other types as well? Wouldn't this be more consistent with the `Part` API and also easy to do from an implementation perspective?

If we decide to do so, the wording in the next paragraph will need adjustments as well.

Reply:
I was satisfied with the original types supported for a first version, but since we added `getContent(Class)`, it seems reasonable to assume other types should be allowed here as well as @chkal points out.

Reply:
Agreed.  I'll add that shortly.